From 40f9cc4b238cdbe107e2d8cedbc4123108ae501a Mon Sep 17 00:00:00 2001
From: Orgad Shaneh <orgads@gmail.com>
Date: Sun, 3 Mar 2024 12:18:31 +0200
Subject: [PATCH 6/6] sim/erc32: Rename EVENT_MAX -> MAX_EVENTS

EVENT_MAX is defined as 0x7FFFFFFF (INT_MAX) in winuser.h, so when
building on Windows, the value is overridden and compilation fails
because the array size of evbuf is too large.

Bug: https://sourceware.org/bugzilla/show_bug.cgi?id=28476
---
 sim/erc32/func.c | 6 +++---
 sim/erc32/sis.h  | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/sim/erc32/func.c b/sim/erc32/func.c
index 85f71817dfd..7fa31813020 100644
--- a/sim/erc32/func.c
+++ b/sim/erc32/func.c
@@ -698,7 +698,7 @@ init_signals(void)
 extern struct disassemble_info dinfo;
 
 struct estate   ebase;
-struct evcell   evbuf[EVENT_MAX];
+struct evcell   evbuf[MAX_EVENTS];
 struct irqcell  irqarr[16];
 
 static int
@@ -867,10 +867,10 @@ init_event(void)
 
     ebase.eq.nxt = NULL;
     ebase.freeq = evbuf;
-    for (i = 0; i < EVENT_MAX; i++) {
+    for (i = 0; i < MAX_EVENTS; i++) {
 	evbuf[i].nxt = &evbuf[i + 1];
     }
-    evbuf[EVENT_MAX - 1].nxt = NULL;
+    evbuf[MAX_EVENTS - 1].nxt = NULL;
 }
 
 void
diff --git a/sim/erc32/sis.h b/sim/erc32/sis.h
index 531ea100184..5bfc4b1cdeb 100644
--- a/sim/erc32/sis.h
+++ b/sim/erc32/sis.h
@@ -31,7 +31,7 @@
 #define I_ACC_EXC 1
 
 /* Maximum events in event queue */
-#define EVENT_MAX	256
+#define MAX_EVENTS	256
 
 /* Maximum # of floating point queue */
 #define FPUQN	1
-- 
2.44.0.windows.1

